### PR TITLE
feat: add new inspect namespaces list command to skaffold

### LINKED ISSUE #8191: [FR] Add flag/config feature to allow users to get renderered manifest or some information about the resulting namespace of render + deploy config
Issues tracks adding flag/config feature to allow users get a renderered manifest or some information about the resulting namespace of render + deploy config.

Option 1 - (Skaffold) Render flag that populates the Namespace configured in deploy stanza
If there is a reason not to change the default behavior for render to support this, then a flag could be introduced on skaffold render to support it.

Option 2 - (Skaffold) Dry run
Introduce the ability to dry run skaffold apply where the output is the resources that would be applied to the cluster. Cloud Deploy could then inspect the dry run resource to extract the namespace that is being used at deploy time. Could be built on top of kubectl apply [dry-run](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#apply) capability.

### PR DESCRIPTION
**Pull request by aaron-prindle**
fixes #8191

### CODE REVIEW

File: cmd/skaffold/app/cmd/inspect_namespaces.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/inspect"
+	namespaces "github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/inspect/namespaces"
+)
+
+func cmdNamespaces() *cobra.Command {
+	return NewCmd("namespaces").
+		WithDescription("View skaffold test information").
```
Comment (renzodavid9) [inline cmd/skaffold/app/cmd/inspect_namespaces.go:33]: I think this is the description from `skaffold inspect tests`, do we need to change it for `skaffold inspect namespaces`?

Comment (aaron-prindle) [reply cmd/skaffold/app/cmd/inspect_namespaces.go]: Good catch, updated now

File: pkg/skaffold/inspect/namespaces/list_test.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+		profiles    []string
+		module      []string
+		err         error
+		expected    string
+	}{
+		{
+			description: "print all deployment namespaces where no namespace is set in manifest(s) or deploy config",
+			manifest:    manifest,
+			expected:    `{"resourceToInfoMap":{"apps/v1, Kind=Deployment":[{"name":"leeroy-app","namespace":"default"}]}}` + "\n",
+			// expected: `` + "\n",
```
Comment (renzodavid9) [inline pkg/skaffold/inspect/namespaces/list_test.go:97]: Can we remove this?

**Review by renzodavid9 — APPROVED** [review]

File: pkg/skaffold/inspect/namespaces/list_test.go (hunk 2 above)
Comment (aaron-prindle) [reply pkg/skaffold/inspect/namespaces/list_test.go:97]: Ah good catch sorry.  Removed now

**Pull request merged**